homework-submission/week2/src/component/Todolist.js

```diff
@@ -0,0 +1,23 @@
+import React, { Component } from "react";
+import Todo from "./Todo";
+
+class Todolist extends Component {
+  render() {
+    const { items } = this.props;
+    const todos = items.map(todo => (
```

Comment:
Actually I spent some extra time to understand hoe this is working.
So What is `todos` or `items` in other words? where it came from?
Answering this question `todos` is an object it came from a `json` has been passed through state then props file that contains an array of objects.

That's right ! But the issue is not with todos, The Issue is how react is working, and by setting an object out sid of the react component is not a solution of making things works, so it may work here and it's working very well I Agree with that, But in the near future you will see that so clearly it's not the way of working with react.

**React is Reacting on notifications that are Coming from state or props**

So how the state or the props are behave, it's working.

in your scenario why is't working right ?! 
well the answer is so simple, you are changing a one singular element and the element is reacting on what you are giving him, so not react it's JSX engine.

So Now you have to follow this in order to understand what is going on...
I need you before you read the point number two I really want you to go where the number one is telling you to go.
 1. State is your component store, and setState is the component notifications manager with props passed from the parent, unless you have something like MOBX or Redux, [GO Here](https://github.com/HackYourFuture/React/pull/204/files#diff-8d80e97daabceb39a2dead2cdd28e3ceR9).
 2. for using `data` what do you have to do is just to change the `handleChange` method 🙂.
 3. now we know from where the problem started and we should also know where it's [going!!]https://github.com/HackYourFuture/React/pull/203/files#diff-dd04c95323f0527b8ead2f8aa8237f4bR18)